### PR TITLE
fix: advanced inference recommendation jobs parameters check

### PR DESCRIPTION
**Pull request by jinpengqi**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by gwang111 — APPROVED** [review]

**Review by trajanikant — COMMENTED** [review]
/bot run all

File: src/sagemaker/inference_recommender/__init__.py [hunk 1]
```diff
@@ -12,3 +12,29 @@
 # language governing permissions and limitations under the License.
 """Classes for using Inference Recommender with Amazon SageMaker."""
 from __future__ import absolute_import
+
+
+class Phase:
```
Comment (jbarz1) [inline src/sagemaker/inference_recommender/__init__.py:17]: would it be possible to leave the implementation in the other file and over here just do this?

```
from sagemaker.inference_recommender.inference_recommender_mixin import Phase, ModelLatencyThreshold
```

File: src/sagemaker/inference_recommender/inference_recommender_mixin.py [hunk 2]
```diff
@@ -464,6 +439,14 @@ def _convert_to_resource_limit_json(self, max_tests: int, max_parallel_tests: in
         """Bundle right_size() parameters into a resource limit for Advanced job"""
         if not max_tests and not max_parallel_tests:
             return None
+        if max_tests and not max_parallel_tests:
```
Comment (jbarz1) [inline src/sagemaker/inference_recommender/inference_recommender_mixin.py:442]: would this be simpler?

```
val = {}
if max_tests:
  val["MaxNumberOfTests"] = max_tests
if max_parallel_tests:
  val["MaxParallelOfTests"] = max_parallel_tests
return val
```

**Review by jbarz1 — APPROVED** [review]
LGTM

**Review by trajanikant — COMMENTED** [review]
/bot run all

**Review by claytonparnell — COMMENTED** [review]
/bot run all

**Pull request merged**